sabnzbd/utils/checkdir.py

```diff
@@ -103,6 +103,30 @@ def isFAT(check_dir):
     return FAT
 
 
+def directory_is_writable(mydir, myfilename):
+    filename = os.path.join(mydir, myfilename)
+    if os.path.exists(filename):
+        try:
+            os.remove(filename)
+        except:
+            return False
+    try:
+        with open(filename, "w") as f:
+            f.write("Some random content")
+        os.remove(filename)
+        return True
+    except:
+        return False
+
+
+def directory_is_writable_basic(mydir):
+    return directory_is_writable(mydir, "sab_test.txt")
+
+
+def directory_is_writable_special_chars(mydir):
+    return directory_is_writable(mydir, "sab_test \\ bla :: , bla.txt")
```

Comment:
But, this will always fail on Windows... 
![image](https://user-images.githubusercontent.com/5703454/166231830-03f5b86b-5976-47db-8c34-44928a03cfec.png)
See also `sanitize_filename` and `CH_LEGAL_WIN`.

Reply:
Ah, yes. I had tested from Linux (plain Ubuntu and WSL-on-Windows) onto FAT and onto NTFS. From both Linux and WSL to NTFS, I can write those chars (although resulting in strange characters on the filename when viewed from Windows, I now see).

Let me think about that. Maybe detect it, and force-turn on Sanitize_Safe. Or exclude that test if on Windows, because then we already know the outcome and because SAB will already take care of it.

Reply:
PS: "interesting" that CI-test of this PR run OK ... also for the Windows-instance. So in that Windows-test, my PR could write "blabla \\ bla : " to disk?

Reply:
It only throws a warning and doesn't block anything, and as you can see, also on CI it's thrown:

https://github.com/sabnzbd/sabnzbd/runs/6245689918?check_suite_focus=true#step:7:145

Reply:
Ah, yes, so `2022-05-01 08:32:42,389::INFO::[notifier:123] Sending notification: Warning - \\?\D:\a\sabnzbd\sabnzbd\tests\cache\Downloads\incomplete is not writeable with special character filenames. This can cause problems. (type=warning, job_cat=None)`

I'll think about a solution on Windows.

Reply:
Seems we can just skip it, or at least this check, on Windows.

Reply:
now skipping if on windows, or safe_sanitize is on